classes/task/sendfailednotifications_task.php

```diff
@@ -141,9 +143,30 @@ private function sendmail($msg, $subject, $emailto) {
         $from->email = self::DEFAULT_SENDER_EMAIL;
         $from->deleted = 0;
         $from->mailformat = FORMAT_HTML;
-    
-        $messageid = email_to_user($user, $from, $subject, html_to_text($msg), $msg);
-        return $messageid;
+
+        // if any rows haven't been sent before then set flag to send
+        $rowstosend = false;
+        foreach ($failedrows as $row) {
+            $rv = $DB->get_records("logstore_xapi_notif_sent_log", array("failedlogid" => $row->id, "email" => $emailto));
+            if (empty($rv)) {
+                $rowstosend = true;
```

Comment:
if you find a condition that sets `$rowstosend` to true, is there any need to continue iterating over the `$failedrows`?

Reply:
Yes, I can exit the loop with a break; here.